src/liballoc/collections/linked_list.rs

```diff
@@ -878,6 +878,58 @@ impl<T> LinkedList<T> {
         unsafe { self.split_off_after_node(split_node, at) }
     }
 
+    /// Removes the element at the given index and returns it.
+    ///
+    /// This operation should compute in O(n) time.
+    ///
+    /// # Panics
+    /// Panics if at >= len
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(linked_list_remove)]
+    /// use std::collections::LinkedList;
+    ///
+    /// let mut d = LinkedList::new();
+    ///
+    /// d.push_front(1);
+    /// d.push_front(2);
+    /// d.push_front(3);
+    ///
+    /// assert_eq!(d.remove(1), 2);
+    /// assert_eq!(d.remove(0), 3);
+    /// assert_eq!(d.remove(0), 1);
+    /// ```
+    #[unstable(feature = "linked_list_remove", issue = "69210")]
+    pub fn remove(&mut self, at: usize) -> T {
+        let len = self.len();
+        assert!(at < len, "Cannot remove at a nonexistent index");
+
+        if at == 0 {
```

Comment:
I'm interested in the benefits of keeping this. Is this to increase performance if the user removes the first or last item? I assume(⚠️) user's will usually use `pop_front` and `pop_back` unless they are removing from a variable index (which, I assume(⚠️) if it's truly variable, wouldn't often be the first or last index) so I don't know if this will come up often.

My personal preference is to write less code when possible and since (I believe) these two cases should still be handled if these checks are removed - I wonder if the potential improvements are worth it. Have we benchmarked both cases?